game/src/Game/Autoload/CursorManager.gd

```diff
@@ -0,0 +1,172 @@
+		loaded_cursors[cursor_name] = cursor
+
+var mouseOverWindow : bool = false
+
+#Handle queued cursor changes and cursor animations
+func _process(delta) -> void:
+	#only attempt to update the mouse when this wont crash anything
+	if mouseOverWindow:
+		var mouseShape : Input.CursorShape = Input.get_current_cursor_shape()
+
+		for shape in currentCursors.keys():
+			if currentCursors[shape] != queuedCursors[shape]:
+				currentCursors[shape] = queuedCursors[shape]
+				if currentCursors[shape] != null:
+					currentCursors[shape].set_hardware_cursor(0, shape)
+				else:
+					Input.set_custom_mouse_cursor(null, shape)
+
+		#The mouse's cursor shape changed (something like we started hovering over text)
+		# reset the current cursor's frame, then switch the active cursor
+		if mouseShape != activeShape:
+			#Current mouse type changed, need to make sure that if the cursor of this new type
+			# is animated, we are providing its frames instead of the frames of the previous active cursor
+			activeShape = mouseShape
+			activeCursor = currentCursors.get(activeShape, null)
+			if activeCursor != null:
+				activeCursor.reset()
+				activeCursor.set_hardware_cursor(0, activeShape)
+			else:
+				Input.set_custom_mouse_cursor(null, activeShape)
+
+		#if we didnt change cursors and are animated, do an update
+		elif activeCursor != null and activeCursor.is_animated:
+			activeCursor._process_cursor(delta,activeShape)
+
+
+func set_prefered_res(res_in:Vector2) -> void:
+	preferred_res = res_in
+
+func set_compat_cursor(cursor_name:StringName, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+	if cursor_name in loaded_cursors:
+		var cursor : CompatCursor = loaded_cursors[cursor_name]
+		cursor.set_resolution(preferred_res)
+		queuedCursors[shape] = cursor
+	else:
+		push_warning("Cursor name %s is not among loaded cursors" % cursor_name)
+		queuedCursors[shape] = null
+
+#NOTE: Each cursor has a corresponding "shape"
+# to indicate when window is busy, normal, doing a drag-select, etc. 
+# You can set this per Control Node under Mouse > Default Cursor Shape 
+
+# set_compat_cursor makes the named vic2 cursor the presently active
+# one for the shape it is currently associated with. By default a cursor
+# is associated with Input.CURSOR_ARROW, but you can override this with the second
+# argument. Use set_compat_cursor as you find it used here in initial_cursor_setup().
+
+func initial_cursor_setup() -> void:
+	set_prefered_res(Vector2(48,48))
+	load_cursors()
+
+	set_compat_cursor(&"normal")
+	# When hovered over a control node with mouse shape set to "busy" (ie. loading screens)
+	#  use the pocket watch cursor
+	set_compat_cursor(&"busy", Input.CURSOR_BUSY)
+
+# To safely change the mouse cursor, the mouse must be over the window
+# this function helps ensure we do it safely
+func _notification(what) -> void:
+	match(what):
+		NOTIFICATION_WM_MOUSE_ENTER:
+			mouseOverWindow = true
+		NOTIFICATION_WM_MOUSE_EXIT:
+			mouseOverWindow = false
```

Comment:
Probably should be connected Window.mouse_entered and Window.mouse_exited in a `_ready()` method, for example:
```gdscript
func _ready() -> void:
  var window : Window = get_window()
  window.mouse_entered.connect(func() -> void: is_mouse_over_window = true)
  window.mouse_exited.connect(func() -> void: is_mouse_over_window = false)
```